lib/parse.js

```diff
@@ -83,6 +83,10 @@ internals.parseKeys = function (key, val, options) {
         return;
     }
 
+    // Transform dot notation to bracket notation
+
+    key = key.replace(/\.(\w+)/g, '[$1]');
```

Comment:
This feels too strict, `\w` is alphanumeric plus underscores, so it rules out plenty of valid characters. Maybe better to make it `/\.([^\.\[]+)/`.

I'm also curious how big of a hit this has on performance..